### ISSUE TITLE
X axis don't show full text

### ISSUE THREAD

**Issue by prsolucoes** [apexcharts/apexcharts.js#678]
Hi,

The x axis don't show full text. Example comparing apex chart (on middle) to chart.js (on left):

![image](https://user-images.githubusercontent.com/395096/59562664-ae98da80-9005-11e9-8617-5279eca98756.png)

Can anyone help me?

Thanks.

**Comment by junedchhipa** [comment]
Can you please provide a full codepen of the demo?
There might be some misconfiguration in the options object.

**Comment by prsolucoes** [comment]
Hi,

It is working with the server data, but the chart has a height of 200 with labels from 1900 to 1910, only this.

![image](https://user-images.githubusercontent.com/395096/59562986-83fd5080-900a-11e9-848d-cfde81fa71d2.png)

I make it working setting the min-height for labels = 40. There is any automatic way to don't need set min-height?

```
xaxis: {
          type: "category",
          categories: [1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998],
          labels: {
            show: true,
            rotate: -45,
            rotateAlways: true,
            hideOverlappingLabels: true,
            trim: true,
            minHeight: 40
          }
        }
```

Thanks.

**Comment by junedchhipa** [comment]
Yeah, it should automatically set the height of x-axis based on label height.
If you can provide a working codepen example, it will be easier for me to fix it.
Or, paste your full configuration of the chart.

**Comment by prsolucoes** [comment]
My project will be open source, but now it is locally, but the configuration is simple and you can reproduce it with my configuration:

```
this.apexChartData = [
  {
    name: "Temperature",
    data: response.data.data.data
  }
];

this.apexChartOptions = {
  ...this.apexChartOptions,
  ...{
    xaxis: {
      type: "category",
      categories: response.data.data.labels,
      labels: {
        show: true,
        rotate: -45,
        rotateAlways: true,
        hideOverlappingLabels: true,
        trim: true,
        minHeight: 40,
        showDuplicates: false
      }
    }
  }
};
```

1 - The chart has a container with height of 300px.

2 - **response.data.data.data** is array:
```
[21.0,33.0,26.0,31.0,26.0,38.0,34.0,38.0,29.0,36.0,35.0,22.0,27.0,22.0,27.0,30.0,31.0,39.0,28.0,39.0,39.0,32.0,32.0,26.0,20.0]
```

3 - **response.data.data.labels** is array:
```
["13:17:06","13:17:16","13:17:27","13:17:37","13:17:47","13:17:57","13:18:08","13:18:18","13:18:29","13:18:39","13:18:49","13:18:59","13:19:10","13:19:20","13:19:31","13:19:41","13:19:51","13:20:01","13:20:12","13:20:22","13:20:32","13:20:42","13:20:52","13:21:03","13:21:13"]
```

**Comment by prsolucoes** [comment]
Other simple example:

![image](https://user-images.githubusercontent.com/395096/59879096-57d22e80-9380-11e9-8d84-168d907ce232.png)

In chart.js all labels are showed and in apex some labels are removed.

**Comment by junedchhipa** [comment]
This is fixed in the latest version of ApexCharts (v3.15.2)

I tried your code and got this result

![Screen Shot 2020-01-24 at 12 08 46 PM](https://user-images.githubusercontent.com/17950663/73049348-4c109280-3ea2-11ea-8c17-50739494711e.png)
